src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/locationsinsideprison/LocationsInsidePrisonClient.kt

```diff
@@ -4,12 +4,14 @@ import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.WebClientResponseException
 import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.locationsinsideprison.model.Location
-import java.util.UUID
 
 class LocationsInsidePrisonClient(private val locationsInsidePrisonApiWebClient: WebClient) {
 
-  fun getLocation(locationId: UUID): Location? = locationsInsidePrisonApiWebClient.get()
-    .uri("/locations/{id}", locationId)
+  fun getLocationsByKeys(keys: List<String>): List<Location> = keys.mapNotNull(::getLocationByKey)
+
+  @Deprecated(message = "We are waiting on a post endpoint to avoid multiple location API calls")
+  private fun getLocationByKey(key: String): Location? = locationsInsidePrisonApiWebClient.get()
```

Comment:
Temporary until work by the locations adds an endpoint to support this.  Note this work has been ticketed up by them.